source/patterns/@aws-solutions-constructs/core/lib/kendra-helper.ts

```diff
@@ -0,0 +1,254 @@
+          ],
+          effect: iam.Effect.ALLOW
+        }),
+        new iam.PolicyStatement({
+          effect: iam.Effect.ALLOW,
+          actions: [
+            "kendra:BatchPutDocument",
+            "kendra:BatchDeleteDocument"
+          ],
+          resources: [
+            targetIndex.attrArn
+          ]
+        }),
+      ]
+    });
+
+    const dataSourceRole: iam.Role = new iam.Role(scope, `data-source-role-${id}`, {
+      assumedBy: new iam.ServicePrincipal('kendra.amazonaws.com'),
+      description: 'Policy for Kendra S3 Data Source',
+      inlinePolicies: {
+        s3CrawlPolicy,
+      },
+    });
+    defaultProps = overrideProps(defaultProps, { roleArn: dataSourceRole.roleArn });
+  }
+
+  const consolidatedProps: kendra.CfnDataSourceProps = consolidateProps(defaultProps, clientProps);
+
+  return new kendra.CfnDataSource(scope, `data-source-${id}`, consolidatedProps);
+
+}
+
+function CreateKendraIndexLoggingRole(scope: Construct, id: string): string {
+  const allowKendraToLogPolicy = new iam.PolicyDocument({
+    statements: [
+      new iam.PolicyStatement({
+        resources: ['*'],
+        actions: [
+          "cloudwatch:PutMetricData"
+        ],
+        effect: iam.Effect.ALLOW,
+        conditions: {
+          StringEquals: {
+            "cloudwatch:namespace": "AWS/Kendra"
+          }
+        }
+      }),
+      new iam.PolicyStatement({
+        resources: [`arn:aws:logs:${Aws.REGION}:${Aws.ACCOUNT_ID}:log-group:/aws/kendra/*`],
+        actions: [
+          "logs:CreateLogGroup"
+        ],
+        effect: iam.Effect.ALLOW,
+      }),
+      new iam.PolicyStatement({
+        resources: [`arn:${Aws.PARTITION}:logs:${Aws.REGION}:${Aws.ACCOUNT_ID}:log-group:/aws/kendra/*`],
+        actions: [
+          "logs:DescribeLogGroups"
+        ],
+        effect: iam.Effect.ALLOW,
+      }),
+      new iam.PolicyStatement({
+        resources: [`arn:${Aws.PARTITION}:logs:${Aws.REGION}:${Aws.ACCOUNT_ID}:log-group:/aws/kendra/*:log-stream:*`],
+        actions: [
+          'logs:CreateLogStream',
+          'logs:PutLogEvents',
+          'logs:DescribeLogStream',
+        ],
+        effect: iam.Effect.ALLOW,
+      }),
+    ],
+  });
+
+  const indexRole: iam.Role = new iam.Role(scope, `kendra-index-role-${id}`, {
+    assumedBy: new iam.ServicePrincipal('kendra.amazonaws.com'),
+    description: 'Allow Kendra index to write CloudWatch Logs',
+    inlinePolicies: {
+      AllowLogging: allowKendraToLogPolicy,
+    },
+  });
+  addCfnSuppressRules(indexRole, [{
+    id: "W11",
+    reason: "PutMetricData does not allow resource specification, " +
+      "scope is narrowed by the namespace condition. " +
+      "https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazoncloudwatch.html"
+  }]);
+  return indexRole.roleArn;
+}
+
+// @summary Confirm each entry is a correct value, uppercase each entry
+export function normalizeKendraPermissions(rawPermissions: string[]): string[] {
```

Comment:
This goes back to my `enum` comment previously. If we force the enum, we shouldn't need this error checking internally.